File: TiaCodegen/Commands/Functions/CTUDCall.cs
using System.Linq;
using TiaCodegen.Commands.Functions.Base;
using TiaCodegen.Enums;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions
{
    public class CTUDCall : SystemFunctionBlockCall
    {
        public CTUDCall(string instanceName,
            IOperationOrSignal cd = null,
            IOperationOrSignal r = null,
            IOperationOrSignal ld = null,
            IOperationOrSignal pv = null,
            IOperationOrSignal qu = null,
            IOperationOrSignal qd = null,
            IOperationOrSignal cv = null) : base("CTUD", instanceName, null)
        {
            Interface["CD"] = new IOperationOrSignalDirectionWrapper(cd, Direction.Input);
            Interface["R"] = new IOperationOrSignalDirectionWrapper(r, Direction.Input);
            Interface["LD"] = new IOperationOrSignalDirectionWrapper(ld, Direction.Input);
            Interface["PV"] = new IOperationOrSignalDirectionWrapper(pv, Direction.Input);
            Interface["QU"] = new IOperationOrSignalDirectionWrapper(qu, Direction.Output);
            Interface["QD"] = new IOperationOrSignalDirectionWrapper(qd, Direction.Output);
            Interface["CV"] = new IOperationOrSignalDirectionWrapper(cv, Direction.Output);

            TemplateValueName = "value_type";
            TemplateValueType = "Type";
            TemplateValue = "Int";
            HasNoEn = true;

            Children.AddRange(Interface.Values.Where(x => x.OperationOrSignal != null).Select(x => x.OperationOrSignal));
        }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/CTUDCall.ts
import { IOperationOrSignal } from '../../Interfaces/IOperationOrSignal.js';
import { IOperationOrSignalDirectionWrapper } from '../../Interfaces/IOperationOrSignalDirectionWrapper.js';
import { Direction } from '../../Enums/Direction.js';
import { SystemFunctionBlockCall } from './Base/SystemFunctionBlockCall.js';

export interface CTUDCallOptions {
    instanceName: string;
    cd?: IOperationOrSignal | null;
    r?: IOperationOrSignal | null;
    ld?: IOperationOrSignal | null;
    pv?: IOperationOrSignal | null;
    qu?: IOperationOrSignal | null;
    qd?: IOperationOrSignal | null;
    cv?: IOperationOrSignal | null;
    templateValue?: string;
}

export class CTUDCall extends SystemFunctionBlockCall {
    constructor(options: CTUDCallOptions) {
        const { instanceName, cd = null, r = null, ld = null, pv = null, qu = null, qd = null, cv = null, templateValue = 'Int' } = options;
        super({ functionName: 'CTUD', instanceName });
        this.iface['CD'] = new IOperationOrSignalDirectionWrapper(cd, Direction.Input);
        this.iface['R'] = new IOperationOrSignalDirectionWrapper(r, Direction.Input);
        this.iface['LD'] = new IOperationOrSignalDirectionWrapper(ld, Direction.Input);
        this.iface['PV'] = new IOperationOrSignalDirectionWrapper(pv, Direction.Input);
        this.iface['QU'] = new IOperationOrSignalDirectionWrapper(qu, Direction.Output);
        this.iface['QD'] = new IOperationOrSignalDirectionWrapper(qd, Direction.Output);
        this.iface['CV'] = new IOperationOrSignalDirectionWrapper(cv, Direction.Output);

        this.templateValueName = 'value_type';
        this.templateValueType = 'Type';
        this.templateValue = templateValue;
        this.hasNoEn = true;

        for (const w of Object.values(this.iface)) {
            if (w.operationOrSignal !== null) this.children.push(w.operationOrSignal);
        }
    }
}
